app/infrastructure/database/repositories/sqlalchemy_speaker_embedding_repository.py

```
@@ -0,0 +1,302 @@
+            await self.session.commit()
+            await self.session.refresh(orm_embedding)
+            logger.info("Speaker embedding added: %s", orm_embedding.uuid)
+            return str(orm_embedding.uuid)
+        except SQLAlchemyError as e:
+            await self.session.rollback()
+            logger.error("Failed to add speaker embedding: %s", str(e))
+            raise DatabaseOperationError(
+                operation="add_speaker_embedding",
+                reason=str(e),
+                original_error=e,
+            )
```

Ruff (0.15.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)